storage/flux/window.go

```diff
@@ -0,0 +1,183 @@
+package storageflux
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+
+	"github.com/apache/arrow/go/arrow/array"
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/arrow"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/values"
+	"github.com/influxdata/influxdb/v2"
+)
+
+// splitWindows will split a windowTable by creating a new table from each
+// row and modifying the group key to use the start and stop values from
+// that row.
+func splitWindows(ctx context.Context, in flux.Table, f func(t flux.Table) error) error {
+	wts := &windowTableSplitter{
+		ctx: ctx,
+		in:  in,
+	}
+	return wts.Do(f)
+}
+
+type windowTableSplitter struct {
+	ctx context.Context
+	in  flux.Table
+}
+
+func (w *windowTableSplitter) Do(f func(flux.Table) error) error {
+	defer w.in.Done()
```

Comment:
Why is this needed? You are calling `w.in.Do()` anyway

Reply:
It's possible for this to fail when reading the schema. Since the call to `Done()` doesn't harm anything and should always happen, this will ensure the table is always marked as processed since that's a requirement for the `Do()` function.